contracts/common/VaultRecoverable.sol

```diff
@@ -27,8 +31,9 @@ contract VaultRecoverable is IVaultRecoverable, EtherTokenConstant, IsContract {
         if (_token == ETH) {
             vault.transfer(address(this).balance);
         } else {
-            uint256 amount = ERC20(_token).balanceOf(this);
-            ERC20(_token).transfer(vault, amount);
+            ERC20 token = ERC20(_token);
+            uint256 amount = token.balanceOf(this);
+            require(ERC20(_token).safeTransfer(vault, amount), ERROR_TOKEN_TRANSFER_FAILED);
```

Comment:
Not sure if we should add this change at this moment—making sure the token transfer was successful is the _right_ thing to do, but it's really only a problem for non-transferable tokens.

Reply:
I think it's fine this way.